checkout-kata.Core/CheckoutMachine.cs

```diff
@@ -93,6 +94,43 @@ private void ProcessAllDiscounts()
             }
         }
 
+        /// <summary>
+        /// Calculates all the discounts that are applicable to the items in the shopping basket.
+        /// </summary>
+        private void ProcessAllDiscounts()
+        {
+            foreach (var item in ShoppingBasket)
+            {
+                int unitPrice = item.Key.UnitPrice;
+
+                int itemCount = item.Value;
+
+                while (item.Key.Discounts.Count > 0)
+                {
+                    var currentDiscount = item.Key.Discounts.MaxBy(x => x.QuantityNeededForDiscount)
```

Comment:
This is fine as it is but to make the code clearer you could encapsulate the 
.Key.Discounts.MaxBy(x => x.QuantityNeededForDiscount)
qualification into a method onto the Item class,eg,
var currentDiscount= item.LargestDiscount

Reply:
I like this.